src/helper/parseAccountField.ts

```diff
@@ -0,0 +1,31 @@
+import type { FilterQuery } from "mongoose";
+import type { ITransaction } from "../models/transactions";
+
+export function parseAccountField(
+  account: string | string[] | undefined
+): FilterQuery<ITransaction> {
+  const filterQuery: FilterQuery<ITransaction> = {};
+  let i, il, j, jl;
+
+  if (Array.isArray(account) && account.length === 1) {
```

Comment:
This maybe has a low perf impact but still want to mention:
We could theoretically determine if the account is a single string and if it contains three parts. If it contains three parts, we could just search for accounts: account.

Reply:
adapted. It is 6 times faster...